ShopperInsights/src/main/java/com/braintreepayments/api/ShoppingInsightsApiCall.kt

```diff
@@ -0,0 +1,32 @@
+package com.braintreepayments.api
+
+import PaymentMethodDetails
+import PaymentMethods
+import ShopperInsightApiResult
+
+internal class ShoppingInsightsApiCall(
+    private val shoppingInsightsCreateBody: ShoppingInsightsCreateBody
+) {
+    fun execute(request: ShopperInsightsApiRequest): ShopperInsightApiResult {
```

Comment:
```suggestion
    fun fetchShopperInsights(request: ShopperInsightsApiRequest): ShopperInsightApiResult {
```
Or any other descriptive name that makes sense to you all

Reply:
would `fetch` work? Class name already has `ShoppingInsights`

Reply:
Assuming there might be more API calls added to this API client in the future, I think it would be helpful for the method to indicate what API call is actually being made - or do we think this will be the only API call for this feature?

Reply:
I don't feel that strongly about it though 🤷

Reply:
If we need to add more API calls in the future I would assume we would be creating new classes for them.

Reply:
We typically have one API "client" per payment feature that handles all internal API calls

Reply:
What would a good example to look at in our codebase to get a better understanding?

Reply:
[ThreeDSecureAPI ](https://github.com/braintree/braintree_android/blob/main/ThreeDSecure/src/main/java/com/braintreepayments/api/ThreeDSecureAPI.java) provides an example

Reply:
@sarahkoop Thanks! I am concerned about API classes getting bloated if we follow that pattern, especially the unit test files.

Reply:
@tdchow What are your thoughts ?

Reply:
i think having APIs separated into their own class was a pattern that was introduced at the very start of the mobile checkout SDK, I'm not entirely sure what the reasoning was.

most projects I've seen have an API class that contains a logical grouping of multiple API calls. although they'd use a networking library like retrofit. so each function/api call wouldn't have any customized logic. (retrofit demo example: https://github.com/square/retrofit/blob/a4b81ed4ffc40a33c35bb466c4be707749b8c858/samples/src/main/java/com/example/retrofit/SimpleService.java#L39C27-L39C27)

in our case, I'm leaning towards the rename and remaining flexible on the possibility of adding an additional API call to this class.

Reply:
What about a wrapper class that groups all the api method calls into one class for a particular module, similar to the `ThreeDSecureAPI` example. The only difference being that you move each api implementation into their own class file. It would help keep each class and its unit tests smaller in size and easier to read, even if a client needed to make many api calls everything would still continue to be manageable. Thoughts ?

Reply:
That sounds good to me 👍

Reply:
I agree, sounds like a good solution!

Reply:
Cool, fixed